xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js

```diff
@@ -25,13 +25,40 @@ define('modalTranslationKeys', [], [
 
 define('modal', ['jquery', 'l10n!modal', 'bootstrap'], function($, translations) {
   'use strict';
-  var modalTemplate = 
+  // Fetch the cross icon from the icon theme to fill up the modal template.
+  let iconURL = `${XWiki.contextPath}/rest/wikis/${XWiki.currentWiki}/iconThemes/icons?name=cross`;
+  let iconRequest = new XMLHttpRequest();
```

Comment:
any reason to not use a jquery `$.get` call here?

Reply:
No specific reason. I don't know much about async javascript and API calls yet. I'll spend some time to try and use jquery instead :+1:
Setting this PR as a draft until this is done